src/aks-preview/azext_aks_preview/_openai_wrapper.py

```diff
@@ -82,8 +86,70 @@ def run_system_script(script_content: str):
         cmd = ["powershell", "-Command", script_content]
     else:
         cmd = ["bash", "-c", script_content]
-    result = subprocess.run(cmd, text=True)
-    return result.returncode
+    subprocess.run(cmd, text=True)
+    return ""
+
+
+def detect_az_error(az_output):
+    # this is based on how azure cli is logging the error to output (message_details)
+    # https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/aaz/_error_format.py#L78
+    pattern = rb'Code:([^\n]+)\nMessage:([^\x1b]+)\x1b'
+    matches = re.findall(pattern, az_output)
+    if matches:
+        for match in matches:
+            try:
+                code = match[0].decode('utf-8').strip()
+                message = match[1].decode('utf-8').strip()
+                return AZ_ERROR_FORMATTER.format(code, message)
+            except Exception:
+                pass
+    return ""
+
+
+def strip_terminal_escapes(byte_string):
+    # Regular expression pattern to match various terminal escape sequences
+    patterns = [
+        b'\033\[[0-9;]*m',  # Color/style changes
+        b'\033\[[0-9]+[ABCD]',  # Cursor movements
+        b'\033\[[0-9;]+[Hf]',  # Cursor positioning
+        b'\033\[2J',  # Clear entire screen
+        b'\033\[K'  # Clear from cursor to end of line
+    ]
+
+    cleaned = byte_string
+    for pattern in patterns:
+        cleaned = re.sub(pattern, b'', cleaned)
+
+    return cleaned.decode('utf-8', 'replace')
+
+
+def spawn_and_run_system_script(script_content: str):
+    """
+    Note: this doesn't work on Windows because there's no pty support
+    we use spawn to capture child output,
+    so user can still interact with yes/no answer from script
+    """
+    if IS_MS_WINDOWS:
+        cmd = ["powershell", "-Command", script_content]
```

Comment:
Seems this is never reached. Or according to description, should just return/raise exception?

Reply:
make sense, just add raise exception if this func is called from windows